### PR TITLE
Youtube Video Spotlight (play youtube in a keep card)

### PR DESCRIPTION
**Pull request by ahsu1230**
Figure out if the link is in youtube format
then display it with a 3-max line description below it (uses ellipsis directive)

@andrewconner @yipingliao

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/keep/keep.js [hunk 1]
```diff
@@ -82,6 +82,7 @@ angular.module('kifi')
         var useBigLayout = false;
         var strippedSchemeRe = /^https?:\/\//;
         var domainTrailingSlashRe = /^([^\/]*)\/$/;
+        var youtubeLinkRe = /.*(?:youtu.be\/|v\/|u\/\w\/|embed\/|watch\?v=)([^#\&\?]*).*/;
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/keep/keep.js:85]: taken from http://stackoverflow.com/questions/3452546/javascript-regex-how-to-get-youtube-video-id-from-url

File: kifi-backend/modules/shoebox/angular/src/keep/keep.js [hunk 2]
```diff
@@ -114,6 +116,17 @@ angular.module('kifi')
...
+        function isYoutubeCard(url) {
+          var strippedSchemeLen = (url.match(strippedSchemeRe) || [''])[0].length;
+          var match = url.substr(strippedSchemeLen).match(youtubeLinkRe);
+          if (match && match[1]) {
+            scope.isYoutubeCard = true;
+            scope.youtubeId = match[1];
+          }
+          return match;
+        }
+        isYoutubeCard(scope.keep.url);
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/keep/keep.js]: This function call should be moved to the bottom of the controller code, in the "on link" section. Alternatively (and I like this better), @andrewconner has started to make an `init()` function where all the initializing code goes, and invokes `init()` at the bottom of a controller or directive's link code.  For an example, see `search.js`.

**Comment by yipingliao** [comment]
:+1:

File: kifi-backend/modules/shoebox/angular/src/keep/keep.js (hunk 2 above)
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/keep/keep.js]: I see, will do that

**Pull request merged**